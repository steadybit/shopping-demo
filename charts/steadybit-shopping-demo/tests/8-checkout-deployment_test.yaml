templates:
  - 8-checkout-deployment.yml
chart:
  appVersion: v0.0.0
tests:
  - it: manifest should match snapshot
    set:
    asserts:
      - matchSnapshot: { }
  - it: manifest should match snapshot replica count
    set:
      checkout:
        replicaCount: 2
    asserts:
      - matchSnapshot: { }
  - it: manifest should match snapshot with horizontalPodAutoscaler and podDisruptionBudget
    set:
      checkout:
        replicaCount: 2
        horizontalPodAutoscaler: true
        podDisruptionBudget: true
    asserts:
      - matchSnapshot: { }