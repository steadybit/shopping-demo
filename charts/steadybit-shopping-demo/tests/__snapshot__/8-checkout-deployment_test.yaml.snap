manifest should match snapshot:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        domain: shop-checkout
        run: checkout
        service-tier: "1"
        tags.datadoghq.com/service: shop-checkout
        tags.datadoghq.com/version: 1.0.0
      name: checkout
      namespace: steadybit-demo
    spec:
      replicas: 1
      selector:
        matchLabels:
          run: checkout
      strategy:
        rollingUpdate:
          maxSurge: 0
          maxUnavailable: 1
        type: RollingUpdate
      template:
        metadata:
          labels:
            domain: shop-checkout
            run: checkout
            service-tier: "1"
            tags.datadoghq.com/service: shop-checkout
            tags.datadoghq.com/version: 1.0.0
        spec:
          affinity:
            podAntiAffinity:
              requiredDuringSchedulingIgnoredDuringExecution:
                - labelSelector:
                    matchExpressions:
                      - key: run
                        operator: In
                        values:
                          - checkout
                  topologyKey: topology.kubernetes.io/zone
          containers:
            - env:
                - name: SPRING_PROFILES_ACTIVE
                  value: kubernetes
                - name: DD_SERVICE
                  valueFrom:
                    fieldRef:
                      fieldPath: metadata.labels['tags.datadoghq.com/service']
                - name: DD_VERSION
                  valueFrom:
                    fieldRef:
                      fieldPath: metadata.labels['tags.datadoghq.com/version']
              image: steadybit/checkout:develop
              imagePullPolicy: Always
              livenessProbe:
                httpGet:
                  path: /actuator/health/liveness
                  port: 8085
                initialDelaySeconds: 60
                periodSeconds: 10
                timeoutSeconds: 3
              name: checkout
              ports:
                - containerPort: 8085
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /actuator/health/readiness
                  port: 8085
                initialDelaySeconds: 60
                periodSeconds: 10
                timeoutSeconds: 5
              resources:
                limits:
                  cpu: 250m
                  ephemeral-storage: 500Mi
                  memory: 1024Mi
                requests:
                  cpu: 50m
                  ephemeral-storage: 250Mi
                  memory: 128Mi
          imagePullSecrets:
            - name: regcredinternal
          serviceAccountName: steadybit-demo
  2: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        run: checkout
      name: checkout
      namespace: steadybit-demo
    spec:
      ports:
        - port: 8085
          protocol: TCP
          targetPort: 8085
      selector:
        run: checkout
      type: ClusterIP
manifest should match snapshot replica count:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        domain: shop-checkout
        run: checkout
        service-tier: "1"
        tags.datadoghq.com/service: shop-checkout
        tags.datadoghq.com/version: 1.0.0
      name: checkout
      namespace: steadybit-demo
    spec:
      replicas: 2
      selector:
        matchLabels:
          run: checkout
      strategy:
        rollingUpdate:
          maxSurge: 0
          maxUnavailable: 1
        type: RollingUpdate
      template:
        metadata:
          labels:
            domain: shop-checkout
            run: checkout
            service-tier: "1"
            tags.datadoghq.com/service: shop-checkout
            tags.datadoghq.com/version: 1.0.0
        spec:
          affinity:
            podAntiAffinity:
              requiredDuringSchedulingIgnoredDuringExecution:
                - labelSelector:
                    matchExpressions:
                      - key: run
                        operator: In
                        values:
                          - checkout
                  topologyKey: topology.kubernetes.io/zone
          containers:
            - env:
                - name: SPRING_PROFILES_ACTIVE
                  value: kubernetes
                - name: DD_SERVICE
                  valueFrom:
                    fieldRef:
                      fieldPath: metadata.labels['tags.datadoghq.com/service']
                - name: DD_VERSION
                  valueFrom:
                    fieldRef:
                      fieldPath: metadata.labels['tags.datadoghq.com/version']
              image: steadybit/checkout:develop
              imagePullPolicy: Always
              livenessProbe:
                httpGet:
                  path: /actuator/health/liveness
                  port: 8085
                initialDelaySeconds: 60
                periodSeconds: 10
                timeoutSeconds: 3
              name: checkout
              ports:
                - containerPort: 8085
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /actuator/health/readiness
                  port: 8085
                initialDelaySeconds: 60
                periodSeconds: 10
                timeoutSeconds: 5
              resources:
                limits:
                  cpu: 250m
                  ephemeral-storage: 500Mi
                  memory: 1024Mi
                requests:
                  cpu: 50m
                  ephemeral-storage: 250Mi
                  memory: 128Mi
          imagePullSecrets:
            - name: regcredinternal
          serviceAccountName: steadybit-demo
  2: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        run: checkout
      name: checkout
      namespace: steadybit-demo
    spec:
      ports:
        - port: 8085
          protocol: TCP
          targetPort: 8085
      selector:
        run: checkout
      type: ClusterIP
manifest should match snapshot with horizontalPodAutoscaler and podDisruptionBudget:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        domain: shop-checkout
        run: checkout
        service-tier: "1"
        tags.datadoghq.com/service: shop-checkout
        tags.datadoghq.com/version: 1.0.0
      name: checkout
      namespace: steadybit-demo
    spec:
      replicas: 2
      selector:
        matchLabels:
          run: checkout
      strategy:
        rollingUpdate:
          maxSurge: 0
          maxUnavailable: 1
        type: RollingUpdate
      template:
        metadata:
          labels:
            domain: shop-checkout
            run: checkout
            service-tier: "1"
            tags.datadoghq.com/service: shop-checkout
            tags.datadoghq.com/version: 1.0.0
        spec:
          affinity:
            podAntiAffinity:
              requiredDuringSchedulingIgnoredDuringExecution:
                - labelSelector:
                    matchExpressions:
                      - key: run
                        operator: In
                        values:
                          - checkout
                  topologyKey: topology.kubernetes.io/zone
          containers:
            - env:
                - name: SPRING_PROFILES_ACTIVE
                  value: kubernetes
                - name: DD_SERVICE
                  valueFrom:
                    fieldRef:
                      fieldPath: metadata.labels['tags.datadoghq.com/service']
                - name: DD_VERSION
                  valueFrom:
                    fieldRef:
                      fieldPath: metadata.labels['tags.datadoghq.com/version']
              image: steadybit/checkout:develop
              imagePullPolicy: Always
              livenessProbe:
                httpGet:
                  path: /actuator/health/liveness
                  port: 8085
                initialDelaySeconds: 60
                periodSeconds: 10
                timeoutSeconds: 3
              name: checkout
              ports:
                - containerPort: 8085
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /actuator/health/readiness
                  port: 8085
                initialDelaySeconds: 60
                periodSeconds: 10
                timeoutSeconds: 5
              resources:
                limits:
                  cpu: 250m
                  ephemeral-storage: 500Mi
                  memory: 1024Mi
                requests:
                  cpu: 50m
                  ephemeral-storage: 250Mi
                  memory: 128Mi
          imagePullSecrets:
            - name: regcredinternal
          serviceAccountName: steadybit-demo
  2: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        run: checkout
      name: checkout
      namespace: steadybit-demo
    spec:
      ports:
        - port: 8085
          protocol: TCP
          targetPort: 8085
      selector:
        run: checkout
      type: ClusterIP
  3: |
    apiVersion: autoscaling/v1
    kind: HorizontalPodAutoscaler
    metadata:
      name: checkout-hpa
      namespace: steadybit-demo
    spec:
      maxReplicas: 3
      minReplicas: 2
      scaleTargetRef:
        apiVersion: apps/v1
        kind: Deployment
        name: checkout
      targetCPUUtilizationPercentage: 200
  4: |
    apiVersion: policy/v1
    kind: PodDisruptionBudget
    metadata:
      name: checkout-pdb
      namespace: steadybit-demo
    spec:
      minAvailable: 1
      selector:
        matchLabels:
          run: checkout
